### PR TITLE
Subs delete

### PR DESCRIPTION
**Pull request by a-alle**

### CODE REVIEW

File: packages/graphql/src/schema/subscriptions/publish-events-to-plugin.ts [hunk 1]
```diff
@@ -77,40 +90,56 @@ function isNodeSubscriptionMeta(event: EventMeta): event is NodeSubscriptionMeta
...
     }
-    return {
-        id: serializeNeo4jValue(event.id),
-        ...extraFields,
-        timestamp: serializeNeo4jValue(event.timestamp),
-        event: event.event,
-        properties,
-    } as SubscriptionsEvent; // Casting here because ts is not smart enough to get the difference between create|update|delete
+    if (isRelationshipSubscriptionMeta(event)) {
+        event.fromTypename ??= schemaModel.getEntityByLabels(event.fromLabels || [])?.name;
```
Comment (a-alle) [inline packages/graphql/src/schema/subscriptions/publish-events-to-plugin.ts]: only execute schemaModel.getEntityByLabels if event.fromLabels exists

File: packages/graphql/src/translate/create-delete-and-params.ts [hunk 2]
```diff
@@ -214,17 +219,91 @@ function createDeleteAndParams({
...
                     const nodeToDelete = `${variableName}_to_delete`;
+
                     res.strs.push(
-                        `WITH ${[...withVars, `collect(DISTINCT ${variableName}) as ${nodeToDelete}`].join(", ")}`
+                        `WITH ${[...withVars, `collect(DISTINCT ${variableName}) as ${nodeToDelete}`].join(
+                            ", "
+                        )}${withRelationshipStr}`
                     );
+                    // ------------------------------------------
+                    /*
```
Comment (a-alle) [inline packages/graphql/src/translate/create-delete-and-params.ts]: rm comment

File: packages/graphql/src/translate/create-delete-and-params.ts [hunk 3]
```diff
@@ -237,6 +316,20 @@ function createDeleteAndParams({
...
+                            `WITH ${filterMetaVariable(withVars).join(", ")}, collect(distinct meta) as update_meta`
+                        );
+                        res.strs.push(
+                            `WITH ${filterMetaVariable(withVars).join(
+                                ", "
+                            )}, REDUCE(m=[], n in update_meta | m+n) as meta`
+                        );
+                    }
+
+                    // ------------------------------------------
```
Comment (a-alle) [inline packages/graphql/src/translate/create-delete-and-params.ts]: rm comment

File: packages/graphql/src/translate/subscriptions/create-connection-event-meta.ts [hunk 4]
```diff
@@ -27,8 +27,12 @@ type EventMetaParameters = {
     typename: string;
     fromVariable: string;
     toVariable: string;
-    fromTypename: string;
-    toTypename: string;
+    fromTypename?: string;
+    toTypename?: string;
+    fromLabels?: string;
```
Comment (a-alle) [inline packages/graphql/src/translate/subscriptions/create-connection-event-meta.ts]: consider making `EventMetaParameters` a union type

File: packages/graphql/src/translate/translate-delete.ts [hunk 5]
```diff
@@ -82,6 +83,10 @@ export function translateDelete({ context, node }: { context: Context; node: Nod
         };
     }
 
+    if (context.subscriptionsEnabled && !deleteInput) {
+        deleteStr = findNodesThatDisconnect(varName);
```
Comment (a-alle) [inline packages/graphql/src/translate/translate-delete.ts]: `findConnectedNodesCypherQuery`

File: packages/graphql/src/translate/translate-delete.ts [hunk 6]
```diff
@@ -109,3 +114,40 @@ function getDeleteReturn(context: Context): Array<string> {
...
+            fromLabels: `labels(${fromVariable})`,
+            toLabels: `labels(${toVariable})`,
+            toProperties: `properties(${toVariable})`,
+            fromProperties: `properties(${fromVariable})`,
+        });
+    return [
+        `CALL {`,
+        `\tWITH ${varName}`,
+        `\tOPTIONAL MATCH (${varName})-[r]-()`,
+        `\tWITH ${varName}, collect(DISTINCT r) as relationships_to_delete`,
```
Comment (a-alle) [inline packages/graphql/src/translate/translate-delete.ts]: AS, THEN, CASE WHEN, END, IN

File: packages/graphql/src/translate/translate-delete.ts [hunk 7]
```diff
@@ -109,3 +114,40 @@ function getDeleteReturn(context: Context): Array<string> {
           ]
         : [];
 }
+
+function findNodesThatDisconnect(varName: string): string {
+    const makeEventWithMetaStr = ({ relVariable, fromVariable, toVariable }) =>
```
Comment (a-alle) [inline packages/graphql/src/translate/translate-delete.ts]: `createDisconnectEventMetaForDeletedNode`

File: packages/graphql/src/schema/subscriptions/publish-events-to-plugin.ts [hunk 8]
```diff
@@ -24,22 +24,48 @@ import type {
...
+                    }
+                }
+            }
+            return events;
+        }, []);
+        const serializedEventsWithoutDuplicates = removeDuplicateEvents(serializedEvents, "disconnect", "delete");
         for (const subscriptionsEvent of serializedEventsWithoutDuplicates) {
             try {
                 const publishPromise = plugin.publish(subscriptionsEvent); // Not using await to avoid blocking
+                console.log("publish!", subscriptionsEvent);
```
Comment (angrykoala) [inline packages/graphql/src/schema/subscriptions/publish-events-to-plugin.ts]:
```suggestion
```

**Pull request merged**